src/api/spec/controllers/webui/projects/public_key_controller_spec.rb

```diff
@@ -27,7 +27,8 @@
     context 'with a project that has no public key' do
       let(:keyinfo_response) { '<keyinfo />' }
 
-      it { expect(response.status).to eq(404) }
+      it { is_expected.to redirect_to(project_show_path(project)) }
+      it { expect(flash[:error]).not_to be_nil }
```

Comment:
@bgeuken 
>  if @Ana06 wouldn't have requested changes, I would have merged this one now.

You can merge it. I still think we should test the error message, but it seems that we won't agree on that and it is not that important.